java-extensions/jni-connector/src/main/java/com/starrocks/jni/connector/ColumnType.java

```diff
@@ -0,0 +1,268 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.jni.connector;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ColumnType {
+
+    public enum TypeValue {
+        BYTE,
+        BOOLEAN,
+        SHORT,
+        INT,
+        FLOAT,
+        LONG,
+        DOUBLE,
+        STRING,
+        DATE,
+        DECIMAL,
+        ARRAY,
+        MAP,
+        STRUCT,
+    }
+
+    TypeValue typeValue;
+    List<String> childNames;
+    List<ColumnType> childTypes;
+    List<Integer> structFieldIndex;
+
+    private final static Map<String, TypeValue> primitiveTypeValueMapping = new HashMap<>();
+    private final static Map<TypeValue, Integer> primitiveTypeValueSize = new HashMap<>();
+
+    static {
+        primitiveTypeValueMapping.put("byte", TypeValue.BYTE);
+        primitiveTypeValueMapping.put("bool", TypeValue.BOOLEAN);
+        primitiveTypeValueMapping.put("short", TypeValue.SHORT);
+        primitiveTypeValueMapping.put("int", TypeValue.INT);
+        primitiveTypeValueMapping.put("float", TypeValue.FLOAT);
+        primitiveTypeValueMapping.put("bigint", TypeValue.LONG);
+        primitiveTypeValueMapping.put("double", TypeValue.DOUBLE);
+        primitiveTypeValueMapping.put("string", TypeValue.STRING);
+        primitiveTypeValueMapping.put("date", TypeValue.DATE);
+        primitiveTypeValueMapping.put("decimal", TypeValue.DECIMAL);
+
+        primitiveTypeValueSize.put(TypeValue.BYTE, 1);
+        primitiveTypeValueSize.put(TypeValue.BOOLEAN, 1);
+        primitiveTypeValueSize.put(TypeValue.SHORT, 2);
+        primitiveTypeValueSize.put(TypeValue.INT, 4);
+        primitiveTypeValueSize.put(TypeValue.FLOAT, 4);
+        primitiveTypeValueSize.put(TypeValue.LONG, 8);
+        primitiveTypeValueSize.put(TypeValue.DOUBLE, 8);
+    }
+
+    static final class StringScanner {
+        String s;
+        int offset;
+
+        StringScanner(String s) {
+            this.s = s;
+            offset = 0;
+        }
+
+        int indexOf(char ch0, char ch1, char ch2) {
```

Comment:
use variable-length arguments here

int indexOf(char ...charArgs) {
    for (char c: charArgs) {
    ...
    }
}